hack/manifestlint.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+set -eux
+
+IS_CONTAINER=${IS_CONTAINER:-false}
+CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-podman}"
+
+# --strict: Disallow additional properties not in schema.
+# --ignore-missing-schemas: Skip validation for resource 
+# definitions without a schema. This will skip the checks
+# for the Custom Resource Definitions(CRDs).
+# -d, --directories strings: A comma-separated list of
+# directories to recursively search for YAML documents.
+# -i, --ignored-filename-patterns strings: A comma-separated
+# list of regular expressions specifying filenames to ignore.
+# -o, --output string: The format of the output of this script.
+# Options are: [stdout json tap].
+
+# We are skipping validation for the files that
+# matches our regexp pattern (i.e. kustom, patch). 
+
+if [ "${IS_CONTAINER}" != "false" ]; then
+    kubeval --strict --ignore-missing-schemas \
+    -d config,examples -i kustom,patch -o tap
+else
+  "${CONTAINER_RUNTIME}" run --rm \
+    --env IS_CONTAINER=TRUE \
+    --volume "${PWD}:/workdir:ro,z" \
+    --entrypoint sh \
+    --workdir /workdir \
+    garethr/kubeval:latest \
```

Comment:
Is this image on docker hub? Is it going to cause us rate limiting problems?

Reply:
Yes, currently it is in Docker hub. After this question, I realized that are more images that we pull from the Docker hub, apart from what we already listed in the [issue](https://github.com/metal3-io/project-infra/issues/166). So, if I check all the other scripts under the hack/, I can see that we are also pulling:
- registry.hub.docker.com/securego/gosec:latest
- registry.hub.docker.com/pipelinecomponents/markdownlint:latest
- registry.hub.docker.com/koalaman/shellcheck-alpine:stable

Since I haven't seen yet a rate limit issue for the markdown, shellcheck, gosec and etc jobs, I would assume it is going to be the same for this script too. But, it could be that we didn't have yet too many PRs at a time, and that's why we didn't face that issue yet. So, probably we will need to mirror/copy those images too if we are going with that path.

Reply:
I saw you added those to the list in the tracking issue, so I think it's OK to say we will handle mirroring all of those images as one project. I suspect we're not seeing rate limiting errors on those jobs because the CI cluster has the relevant images cached.

Reply:
Ah, true. Cashing could be the reason.